--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -317,9 +317,14 @@
 bool ksu_is_manager_apk(char *path)
 {
 #ifdef CONFIG_KSU_SUSFS
-	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH) ||
-			check_v2_signature(path, 384, "7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4")); // 5ec1cff
+	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH)
+	|| check_v2_signature(path, 0x033b, "c371061b19d8c7d7d6133c6a9bafe198fa944e50c1b31c9d8daa8d7f1fc2d2d6") // KernelSU
+	|| check_v2_signature(path, 384, "7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4") // 5ec1cff
+	|| check_v2_signature(path, 0x3e6, "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7") // ksu-next
+	|| check_v2_signature(path, 0x0316, "d6ac81d5a531e3abe4831a44bc662fad30c269cc3ace704342cb5eef742e6722") // rsuntk
+	|| check_v2_signature(path, 0x0394, "ed880907e50c9fc2870594160b9c2720c917274579ce43c48ff19b92ffa1b81f") // backslashxx
+	);
 #else
-	return check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH);
+	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH);
 #endif
 }
\ No newline at end of file
